w3af/plugins/grep/expect_ct.py

```diff
@@ -0,0 +1,94 @@
+"""
+expect_ct.py
+
+Copyright 2019 Andres Riancho
+
+This file is part of w3af, http://w3af.org/ .
+
+w3af is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation version 2 of the License.
+
+w3af is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with w3af; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+"""
+from w3af.core.data.kb.info import Info
+from w3af.core.data.kb.info_set import InfoSet
+from w3af.core.controllers.plugins.grep_plugin import GrepPlugin
+
+ECT_HEADER = 'Expect-CT'
+MAX_REPORTS = 50
+
+
+class expect_ct(GrepPlugin):
+    """
+    Check if HTTPS responses have the Expect-CT header set.
+
+    """
+    def __init__(self):
+        super(expect_ct, self).__init__()
+        self._reports = 0
+
+    def grep(self, request, response):
+        """
+        Check if HTTPS responses have the Expect-CT header set.
+
+        :param request: The HTTP request object.
+        :param response: The HTTP response object
+        :return: None, all results are saved in the kb.
+        """
+        if self._reports > MAX_REPORTS:
```

Comment:
This check might not be required, see https://github.com/andresriancho/w3af/blob/3a1b04fd5a519f6e981d714350b234061cdc38e9/w3af/core/data/kb/info_set.py#L85